sdk/core/azure-core/src/main/java/com/azure/core/util/CoreUtils.java

```diff
@@ -488,4 +488,24 @@ public static long extractSizeFromContentRange(String contentRange) {
 
         return Long.parseLong(sizeString);
     }
+
+    /**
+     * Utility method for parsing query parameters one-by-one without the use of string splitting.
+     * <p>
+     * This method provides an optimization over parsing query parameters with {@link String#split(String)} or a
+     * {@link java.util.regex.Pattern} as it doesn't allocate any arrays to maintain values, instead it parses the
+     * query parameters linearly.
+     * <p>
+     * Query parameter parsing works the following way, {@code key=value} will turn into an immutable {@link Map.Entry}
+     * where the {@link Map.Entry#getKey()} is {@code key} and the {@link Map.Entry#getValue()} is {@code value}. For
+     * query parameters without a value, {@code key=} or just {@code key}, the value will be an empty string.
+     *
+     * @param queryParameters The query parameter string.
+     * @return An {@link Iterator} over the query parameter key-value pairs.
+     */
+    public static Iterator<Map.Entry<String, String>> parseQueryParameters(String queryParameters) {
```

Comment:
We should return the `Map<String, String>` instead of the iterator and the implementation could be a LinkedHashMap to preserve the order.

Reply:
This API would need to change to `Map<String, QuertyParameter>` or `Map<String, List<String>>` in this case, the current design is to iterate over all parsed query parameters and let the caller handle using or storing them.